process/smartContract/processorV2/processV2.go

```diff
@@ -825,27 +846,50 @@ func (sc *scProcessor) doExecuteBuiltInFunction(
 	tx data.TransactionHandler,
 	acntSnd, acntDst state.UserAccountHandler,
 ) (vmcommon.ReturnCode, error) {
-	returnCode, vmInput, txHash, err := sc.prepareExecution(tx, acntSnd, acntDst, true)
+	failureContext := NewFailureContext(sc)
+	retCode, err := sc.doExecuteBuiltInFunctionWithoutFailureProcessing(tx, acntSnd, acntDst, failureContext)
+	if failureContext.processFail {
+		failureProcessingError := sc.processIfErrorWithAddedLogs(
+			acntSnd,
+			tx,
+			failureContext)
+		if failureProcessingError != nil {
+			err = failureProcessingError
+		}
+	}
+	return retCode, err
+}
+
+func (sc *scProcessor) doExecuteBuiltInFunctionWithoutFailureProcessing(
+	tx data.TransactionHandler,
+	acntSnd, acntDst state.UserAccountHandler,
+	failureContext *failureContext,
+) (vmcommon.ReturnCode, error) {
+	returnCode, vmInput, txHash, err := sc.prepareExecution(tx, acntSnd, acntDst, true, failureContext)
 	if err != nil || returnCode != vmcommon.Ok {
 		return returnCode, err
 	}
 
-	snapshot := sc.accounts.JournalLen()
+	failureContext.makeSnaphot()
+	failureContext.
+		setTxHash(txHash).
+		setGasLocked(vmInput.GasLocked)
 
 	var vmOutput *vmcommon.VMOutput
 	vmOutput, err = sc.resolveBuiltInFunctions(vmInput)
 	if err != nil {
 		log.Debug("processed built in functions error", "error", err.Error())
-		return 0, err
+		return vmcommon.Ok, err
 	}
 
 	acntSnd, err = sc.reloadLocalAccount(acntSnd)
 	if err != nil {
-		return 0, err
+		return vmcommon.Ok, err
 	}
 
 	if vmInput.ReturnCallAfterError && vmInput.CallType != vmData.AsynchronousCallBack {
-		return sc.finishSCExecution(make([]data.TransactionHandler, 0), txHash, tx, vmOutput, 0)
+		returnCode, err := sc.finishSCExecution(make([]data.TransactionHandler, 0), txHash, tx, vmOutput, 0)
```

Comment:
why this change?

Reply:
+1

Reply:
at some point there was some code between that and return :) ofc, fixed